### PR TITLE
Removes multiple unmaintained (spark and parallelwrapper related) modules

### PR DESCRIPTION
**Pull request by agibsonccc**
## What changes were proposed in this pull request?

Removes nd4j-aeron
Removes parallel-wrapper training
Removes nd4j-kryo
Removes datavec-spark

## How was this patch tested?
Ensured compilation of of left over modules.
## Quick checklist

The following checklist helps ensure your PR is complete:

- [ X] Eclipse Contributor Agreement signed, and signed commits - see [IP Requirements](https://deeplearning4j.konduit.ai/multi-project/how-to-guides/contribute/eclipse-contributors) page for details
- [ X] Reviewed the [Contributing Guidelines](https://github.com/eclipse/deeplearning4j/blob/master/CONTRIBUTING.md) and followed the steps within.
- [ X] Created tests for any significant new code additions.
- [ X] Relevant tests for your changes are passing.

### CODE REVIEW

File: nd4j/nd4j-parameter-server-parent/nd4j-parameter-server/src/main/java9/module-info.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-open module nd4j.parameter.server {
+open module nd4j.profiler {
```
Comment (treo) [inline nd4j/nd4j-parameter-server-parent/nd4j-parameter-server/src/main/java9/module-info.java]: This looks wrong. I guess you applied that change to the wrong file by mistake and as you were removing a lot of things, this got lost?

File: deeplearning4j/pom.xml [hunk 2]
```diff
@@ -58,6 +57,7 @@
         <module>deeplearning4j-zoo</module>
         <module>deeplearning4j-data</module>
         <module>deeplearning4j-common-tests</module>
+        <module>deeplearning4j-parallelwrapper</module>
```
Comment (treo) [inline deeplearning4j/pom.xml:60]: Parallelwrapper is its own top level module now?

**Review by treo — CHANGES_REQUESTED** [review]

File: nd4j/nd4j-parameter-server-parent/nd4j-parameter-server/src/main/java9/module-info.java (hunk 1 above)
Comment (agibsonccc) [reply nd4j/nd4j-parameter-server-parent/nd4j-parameter-server/src/main/java9/module-info.java]: It's a removed module anyways I guess we should be good to go?

File: deeplearning4j/pom.xml (hunk 2 above)
Comment (agibsonccc) [reply deeplearning4j/pom.xml:60]: Yes since scaleout doesn't exist. The directory layout doesn't affect users.

**Review by treo — APPROVED** [review]

**Pull request merged**